tests/scripts/all.sh

```diff
@@ -580,6 +581,10 @@ msg "test/build: key-exchanges (gcc)" # ~ 1 min
 cleanup
 record_status tests/scripts/key-exchanges.pl
 
+msg "build: Unix make, gcc and g++ test" # ~ 30s
+cleanup
+make TEST_CPP=1
```

Comment:
Should we merge this with an existing build?

Advantage: faster.

Downside: slightly less good reports if it fails since we'd have to look closer at the logs to see if it's the fault of the C++. But this is `all.sh` anyway, we don't.

I'm not suggesting to add `TEST_CPP=1` to an existing command line, that would make it harder to backport this  script. Rather, add this after an existing `make`, before a subsequent `cleanup`.

Reply:
So you'd like to have the following code below:
```
 msg "build: Unix make, -Os (gcc)" # ~ 30s
 cleanup
 make CC=gcc CFLAGS='-Werror -Wall -Wextra -Os'
+make TEST_CPP=1
```
?

Reply:
Yes, that's fine with me.

Reply:
Updated, with additional logs to make things clear.